src/domains/chart/components/chart-with-loader/chart-with-loader.tsx

```diff
@@ -281,7 +281,9 @@ export const ChartWithLoader = ({
 
   useSelector(selectSpacePanelTransitionEndIsActive)
 
-  const [selectedDimensions, setSelectedDimensions] = useState<string[]>([])
+  // eslint-disable-next-line max-len
+  const selecteddimensionsMapped = attributes?.selectedDimensions?.filter((dim) => dim.correlationScore !== undefined).map((dim) => dim.name) || []
+  const [selectedDimensions, setSelectedDimensions] = useState<string[]>(selecteddimensionsMapped)
```

Comment:
does low level `chart-with-loader` needs to know about `correlationScore`? Can we have those conditionals and here use just lower level `selectedDimensions` as strings or array of strings?

Reply:
What I'm doing here with correlationScores is play safe by just returning an empty array (same as it was before) in case there is no correlation score.
another issue to have in mind here is that in a future iteration I will need to display these scores on the charts. So I'd prefer using the whole dimension instead of just an array of strings. Not 100% sure about these though...

Reply:
I am trying to have dashboard working in more lower-level, so it doesn't grow that much, it's already quite complicated. For example we could have the "selectedDimensions" which can be used in many different ways (ie. for other features than correlationScores as well), and if you want to display some new elements in charts (in the future) - we can have just one prop like "customElement" (we already have `customElementForDygraph` for example)

Reply:
I removed the filtering from the array since now its part of the attributes overrides.
However I retains the Dimension[] struct (instead of array of strings) since there will be some addition info required to be shown there soon. 
More importantly it doesn't make much difference if the mapping happens on the parent component or here I think (let me know if this is the case and I'll make the mapping above it)
Please note if there are no selectedDimension it directly defaults to empty array (no mapping is done...)

Reply:
@VLegakis the `attributes` object structure is something like our public, low level Dashboard's API. For example, people can put those attributes as `data-something=...` DOM attributes, in static custom dashboards (html files). So i think this `attributes` object should stay as much abstract as possible. We should also document in the future those settings in `https://learn.netdata.cloud/docs/agent/web/gui/custom#selecting-dimensions` page. If we'll have the new structure, we will need to write in docs:
> If you need to have some dimensions pre-selected, add selected dimensions in following format: `data-selected-dimensions="[{name: dimension1, correlationScore: true},...]"`.

Even is someone will not use `correlationScore` at all. That can go out of hand. I understand that we'll need some additional features for `correlationScore`, like displaying some additional info, elements, etc. but for that we could have some additional api (additional properties in attributes), even as reactNode type.

But I also don't want to block you, and I understand that going with the current implementation could be quicker, so if you want to merge it and move forward, please go ahead, we can eventually change it in the future.

Reply:
Got it! I keep on forgetting this aspect of dashboard component. I'll provide a new PR with simple array of strings. We'll see what we'll do once scores are also required there for rendering

Reply:
Done. So, now `Attributes` are like:
selectedDimensions: ["dim1", "dim2", ...] etc
and also 
data-selected-dimensions: ["dim1", "dim2", ...] etc

Reply:
Thanks @VLegakis!